roadart/lib/src/patcher.dart

```diff
@@ -0,0 +1,61 @@
+import 'dart:collection';
+import 'dart:io';
+import 'dart:math';
+import 'dart:typed_data';
+import 'package:image/image.dart';
+
+// Patch the holes in the segmentation mask
+class Patcher {
+  void patch(String inPath, String outPath) {
+    final Uint8List inBytes = File(inPath).readAsBytesSync();
+    final Image mask = decodePng(inBytes)!;
+
+    final colorCount = HashMap<Color, int>();
+    final xToMinY = HashMap<int, int>();
+    final xToMaxY = HashMap<int, int>();
+    final yToMinX = HashMap<int, int>();
+    final yToMaxX = HashMap<int, int>();
+
+    final int w = mask.width;
+    final int h = mask.height;
+    Color? positiveColor;
+    int maxPositiveCount = 0;
+    for (int x = 0; x < w; ++x) {
+      for (int y = 0; y < h; ++y) {
+        final Color c = mask.getPixel(x, y);
+        colorCount[c] = (colorCount[c] ?? 0) + 1;
+        if ((c.r + c.g + c.b) > 0) {
```

Comment:
nit: consider `if (c.r + c.g + c.b == 0) continue` to reduce the indentation depth

Reply:
done.